keras_cv/layers/preprocessing/vectorized_base_image_augmentation_layer.py

```diff
@@ -0,0 +1,426 @@
+        The resulting images are then stacked back into a ragged image batch.  The
+        behavior of this method should be identical to that of `augment_images()` but
+        is to operate on a batch-wise basis.
+
+        Args:
+            image: a single image from the batch
+            transformation: a single transformation sampled from
+                `get_random_transformations()`.
+            kwargs: all of the other call arguments (i.e. bounding_boxes, labels, etc.).
+        Returns:
+            Augmented image.
+        """
+        raise NotImplementedError(
+            "A ragged image batch was passed to layer of type "
+            f"`{type(self).__name__}`. This layer does not implement "
+            "`augment_ragged_image()`. If this is a `keras_cv`, open a GitHub issue "
+            "requesting Ragged functionality on the layer titled: "
+            f"'`{type(self).__name__}`: ragged image support'. "
+            "If this is a custom layer, implement the `augment_ragged_image()` method."
+        )
+
+    def compute_ragged_image_signature(self, images):
+        """Computes the output image signature for the `augment_image()` function.
+
+        Must be overridden to return tensors with different shapes than the input
+        images.  By default returns either a `tf.RaggedTensorSpec` matching the input
+        image spec, or a `tf.TensorSpec` matching the input image spec.
+        """
+        ragged_spec = tf.RaggedTensorSpec(
+            shape=images.shape[1:],
+            ragged_rank=1,
+            dtype=self.compute_dtype,
+        )
+        return ragged_spec
+
+    def augment_images(self, images, transformations, **kwargs):
+        """Augment a batch of images during training.
+
+        Args:
+          image: 4D image input tensor to the layer. Forwarded from
+            `layer.call()`.  This should generally have the shape [B, H, W, C].
+            Forwarded from `layer.call()`.
+          transformations: The transformations object produced by
+            `get_random_transformations`. Used to coordinate the randomness
+            between image, label, bounding box, keypoints, and segmentation mask.
+
+        Returns:
+          output 4D tensor, which will be forward to `layer.call()`.
+        """
+        raise NotImplementedError()
+
+    def augment_labels(self, labels, transformations, **kwargs):
+        """Augment a batch of  labels during training.
+
+        Args:
+          label: 2D label to the layer. Forwarded from `layer.call()`.
+          transformations: The transformations object produced by
+            `get_random_transformations`. Used to coordinate the randomness
+            between image, label, bounding box, keypoints, and segmentation mask.
+
+        Returns:
+          output 2D tensor, which will be forward to `layer.call()`.
+        """
+        raise NotImplementedError()
+
+    def augment_targets(self, targets, transformations, **kwargs):
```

Comment:
Shouldn't we just drop this, since it's just wrapping around `augment_labels`?

Reply:
Hmm, yeah but thats a follow up PR IMO.  we should remove it from all layers.  francois prefers "targets" because OHE labels are not labels anymore but regression targets - makes a good point but at this point no strong preference.  

In any case follow up.